### PR TITLE
Don't expose EventData fields as Nullable<T>'s

### PR DESCRIPTION
**Pull request by ellismg**
Because we share the EventData type between the consumer and producer
clients, we exposed some fields as Nullable<T>'s if they were assigned
by the Event Hubs service when processing an event (since, when creating
an EventData to send to the service as part of a batch, these properties
would not yet have actual values).

In practice, these properties will always be set when consuming events
and never be set when producing events, while having the Nullable<T>'s
type signiture correctly expresses these semantics, it makes the API
worse to program against in practice. In production scinerios it's
unlikely that you would ever inspect these properties anyway, and in
consumption cases they will alwasy be present so the nullablity doesn't
actually buy you much, but at the expense of `.Value` calls or other
ways of dealing with the (never actually there) `null`s.

So, we change Sequence, Offset and EnqueuedTime to no longer be
nullable. Instead, they are initialized with sentinal values
(`long.MinValue` for `Sequence` and `Offset` and
`default(DateTimeOffset)` for `EnqueuedTime`).  The choice of
`long.MinValue` instead of `0` is intentional, bases on user feedback
from the Track 1 SDKs that zero was a bad sentinal value for these
properties since it could be incorrectly interpreted as semantically
meaningful.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs [hunk 1]
```diff
@@ -89,30 +89,33 @@ public Stream BodyAsStream
...
+        public long SequenceNumber { get; }
 
         /// <summary>
         ///   The offset of the event when it was received from the associated Event Hub partition.
         /// </summary>
         ///
         /// <remarks>
-        ///   This property is only populated for events received from the Event Hubs service.
+        ///   This property is only populated for events received from the Event Hubs service. If this
+        ///   EventData was not recived from the Event Hubs service, the value is long.MinValue.
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs]: nit: Would you be open to considering a `<see cref=` references for these?

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs [hunk 2]
```diff
@@ -89,30 +89,33 @@ public Stream BodyAsStream
...
+        public long Offset { get; }
 
         /// <summary>
         ///   The date and time, in UTC, of when the event was enqueued in the Event Hub partition.
         /// </summary>
         ///
         /// <remarks>
-        ///   This property is only populated for events received from the Event Hubs service.
+        ///   This property is only populated for events received from the Event Hubs service. If this
+        ///   EventData was not recived from the Event Hubs service, the value default(DateTimeOffset).
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs]: nit:  Would you mind wrapping in code tags for consistency with other doc comments?   `<c>default(DateTimeOffset</c>`

**Review by jsquire — COMMENTED** [review]
LGTM.    I left a couple of small requests around doc comments,  if you're open to them.

**Review by jsquire — APPROVED** [review]

**Comment by ellismg** [comment]
> if you're open to them.

Happy to do so!

**Pull request merged**